Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -731,6 +732,60 @@ theorem leftResponse_spec {x : PGame} (h : 0 ≤ x) (j : x.RightMoves) :
   Classical.choose_spec <| (zero_le.1 h) j
 #align pgame.left_response_spec SetTheory.PGame.leftResponse_spec
 
+/-- An explicit upper bound for a family of pre-games, whose left moves are the union of the left
+moves of all the pre-games in the family. -/
+def upperBound {ι : Type u} (f : ι → PGame.{u}) : PGame :=
+  ⟨Σ i, (f i).LeftMoves, PEmpty, fun x => moveLeft _ x.2, PEmpty.elim⟩
```

Comment:
I'm hesitant about this definition. If it's just a helper definition for proving `bddAbove_of_small`, I don't mind it, although I imagine that `⟨ι, PEmpty, f, PEmpty.elim⟩` would be easier to work with. If this is supposed to be a useful canonical definition of an upper bound for pre-games, I think it's a suboptimal one to choose. One desirable property of a canonical upper bound is that it is the least upper bound if one exists. This definition does not satisfy that property. (e.g. upperBound {0, 1/2} = upperBound {{ | }, {0 | 1}} = {0 | } = 1 ≠ 1/2.)

Would it be reasonable to request that this definition and any required lemmas for it be inlined into the proof of `bddAbove_of_small`? (And perhaps the proof can even be golfed by changing the definition to `⟨ι, PEmpty, f, PEmpty.elim⟩`)

Alternatively, if we can think of a simple upper bound definition that does take on the least upper bound when possible, I would be happy to see that used to prove `bddAbove_of_small` instead.